### PR TITLE
Fix eds debounce CommittedUpdates

### PR DESCRIPTION
**Pull request by andrewshaoyu**
**Please provide a description of this PR:**
if `PILOT_ENABLE_EDS_DEBOUNCE` equals false , commitupdate should also be increased.
otherwise pilot server may not be ready.

``` go
func (s *DiscoveryServer) ConfigUpdate(req *model.PushRequest) {
	inboundConfigUpdates.Increment()
	s.InboundUpdates.Inc()
	s.pushChannel <- req
}
```
During `ConfigUpdate` inboundUpdate++,then push model.request,

``` go
push := func(req *model.PushRequest, debouncedEvents int) {
    pushFn(req)
    updateSent.Add(int64(debouncedEvents))
    freeCh <- struct{}{}
}
```

but when enableEDSDebounce, only do `pushFn`.  So InboundUpdates is greater than CommittedUpdates.

``` go
if !opts.enableEDSDebounce && !r.Full {
    // trigger push now, just for EDS
    go pushFn(r)
    continue
}
```

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@andrewshaoyu Could you elaborate the `bug`?

File: pilot/pkg/model/push_context.go
Comment (hzxuzhonghu) [inline pilot/pkg/model/push_context.go:325]: nit: you can make it default 0, and add one when  added to `CommittedUpdates`

Comment (hzxuzhonghu) [reply pilot/pkg/model/push_context.go:325]: So  it  means the debounce merge times, and can get rid of the below unreadable code

**Review by hzxuzhonghu — COMMENTED** [review]
lgtm, i am thinking if you can add a ut

File: pilot/pkg/model/push_context.go
Comment (andrewshaoyu) [reply pilot/pkg/model/push_context.go:325]: yes,i will remove `Default value should be 1.`

**Comment by andrewshaoyu** [comment]
> lgtm, i am thinking if you can add a ut

I have not found the ut about `CommittedUpdates`.  So make a ut which can indicate the bug?

**Review by howardjohn — CHANGES_REQUESTED** [review]
Why not just replace
```
go pushFn(r)
```
with 
```go
go func() {
  pushFn(r)
  updateSent.Add(int64(debouncedEvents))
}
```
?

**Comment by andrewshaoyu** [comment]
``` go
	if !opts.enableEDSDebounce && !r.Full {
		// trigger push now, just for EDS
                go func() {
                  pushFn(r)
                  updateSent.Inc()
                }
		continue
	}
```
If !enableEDSDebounce, pushFn will only commit current PushRequest r, now  just do `updateSent.Inc()`. But How many PushRequests In r ? currently it is one. 

Calculating debounce times when `req.merge(r)`, it will be more clear. We do not care about where to clear debouncedEvents.

I think  how many times the PushRequest was merged is one of PushRequest's Properties. then we can add a metric to expose it.

**Comment by howardjohn** [comment]
> If !enableEDSDebounce, pushFn will only commit current PushRequest r, now just do updateSent.Inc(). But How many PushRequests In r ? currently it is one.

Its always 1, right? So you can do
```
      go func() {
                  pushFn(r)
                  updateSent.Inc()
                }
```

then we don't need all the other changes

**Review by hzxuzhonghu — COMMENTED** [review]
Is the merge right? When two pushrequest merged, the debouncedEvents = 1

**Comment by hzxuzhonghu** [comment]
And updateSent is incremented by one

**Comment by andrewshaoyu** [comment]
now when !enableedsdebounce, pushrequest will be sent right now without merged. so it should be 1.

**Comment by hzxuzhonghu** [comment]
Then how about the other case debounce enabled?

**Comment by andrewshaoyu** [comment]
now using debounceevents as event mered times. If we add a new debounce function， we need to caculate the merged times.

**Review by Xunzhuo — COMMENTED** [review]
LGTM

**Comment by hzxuzhonghu** [comment]
What if two push request merged, the debounceEvents=1

**Comment by hzxuzhonghu** [comment]
IC, req is nil when the first time it runs `req = req.Merge(r)`

**Review by hzxuzhonghu — APPROVED** [review]
LGTM, please add a ut

**Comment by hzxuzhonghu** [comment]
it looks this is not needed any more

**Comment by andrewshaoyu** [comment]
Have been fixed in latest release?

**Comment by hzxuzhonghu** [comment]
I am wrong, nope. 

@howardjohn Another look

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
